src/semantics/expression_builder.ts

```diff
@@ -1513,6 +1511,14 @@ export function newBinaryExprValue(
            And if the type of lvalue and rvalue are both primitive types,
            there is no need to convert the type of lvalue and the type of rvalue to "any".
         */
+        if (left_value.type.kind == ValueTypeKind.ENUM) {
+            const enum_type = left_value.type as EnumType;
+            left_value = newCastValue(enum_type.memberType, left_value);
+        }
+        if (right_value.type.kind == ValueTypeKind.ENUM) {
+            const enum_type = right_value.type as EnumType;
+            right_value = newCastValue(enum_type.memberType, right_value);
+        }
```

Comment:
If `left_value` and `right_value` are both EnumType, they will both be converted to `CastValue`?

Reply:
yes